### PR TITLE
Feat/display food and resources zappy gui

### PR DESCRIPTION
**Pull request by mathieurobert1**
I displayed food and the different resources on tile :
image

### CODE REVIEW

File: gui/include/Render/Render.hpp [hunk 1]
```diff
@@ -78,11 +81,18 @@ class Gui::Render {
 
     private:
 
-        UserCamera                  _camera;    // Camera of the scene.
-        bool                        _isDebug;   // Display or not the debug informations.
-        std::shared_ptr<GameData>   _gameData;  // GameData class to store the game's data.
+        UserCamera                  _camera;            // Camera of the scene.
+        bool                        _isDebug;           // Display or not the debug informations.
```
Comment (Njord201) [inline gui/include/Render/Render.hpp:85]: Add a point at the end of the sentence

**Review by Njord201 — CHANGES_REQUESTED** [review]

File: gui/src/Render/Render.cpp [hunk 2]
```diff
@@ -84,10 +90,93 @@ void Gui::Render::displayDebug(void)
...
-        for (auto &tile : line)
+        for (auto &tile : line) {
             DrawModel(_tileModel, tile.getPositionIn3DSpace(), 0.001f, WHITE);
+            displayFood(tile);
+            displayResources(tile);
+        }
+    }
+}
+
+void Gui::Render::displayFood(Tile tile)
```
Comment (Njord201) [inline gui/src/Render/Render.cpp:104]:
```suggestion
void Gui::Render::displayFood(Tile tile) const
```

File: gui/src/Render/Render.cpp [hunk 3]
```diff
@@ -84,10 +90,93 @@ void Gui::Render::displayDebug(void)
...
+{
+    if (tile._inventory.getFood() == 0)
+        return;
+    Vector3 posTile = tile.getPositionIn3DSpace();
+    Vector3 posFood = POS_FOOD;
+
+    DrawModelEx(_foodModel, (Vector3){posTile.x + posFood.x, posTile.y + posFood.y, posTile.z + posFood.z}, ROTATION_AXIS_FOOD, ROTATION_ANGLE_FOOD, SCALE_FOOD, WHITE);
+}
+
+void Gui::Render::displayResources(Tile tile)
```
Comment (Njord201) [inline gui/src/Render/Render.cpp:114]:
```suggestion
void Gui::Render::displayResources(Tile tile) const
```

File: gui/src/Render/Render.cpp [hunk 4]
```diff
@@ -84,10 +90,93 @@ void Gui::Render::displayDebug(void)
...
+{
+    displayLinemate(tile);
+    displayMendiane(tile);
+    displayPhiras(tile);
+    displaySibur(tile);
+    displayThystame(tile);
+    displayDeraumere(tile);
+}
+
+void Gui::Render::displayLinemate(Tile tile)
```
Comment (Njord201) [inline gui/src/Render/Render.cpp:124]:
```suggestion
void Gui::Render::displayLinemate(Tile tile) const
```

File: gui/src/Render/Render.cpp [hunk 5]
```diff
@@ -84,10 +90,93 @@ void Gui::Render::displayDebug(void)
...
+{
+    Vector3 posTile = tile.getPositionIn3DSpace();
+
+    if (tile._inventory.getLinemate() != 0) {
+        Vector3 posLinemate = POS_LINEMATE;
+        DrawModelEx(_linemateModel, (Vector3){posTile.x + posLinemate.x, posTile.y + posLinemate.y, posTile.z + posLinemate.z}, ROTATION_AXIS_LINEMATE, ROTATION_ANGLE_LINEMATE, SCALE_LINEMATE, WHITE);
+    }
+}
+
+void Gui::Render::displayMendiane(Tile tile)
```
Comment (Njord201) [inline gui/src/Render/Render.cpp:134]:
```suggestion
void Gui::Render::displayMendiane(Tile tile) const
```

File: gui/src/Render/Render.cpp [hunk 6]
```diff
@@ -84,10 +90,93 @@ void Gui::Render::displayDebug(void)
...
+{
+    Vector3 posTile = tile.getPositionIn3DSpace();
+
+    if (tile._inventory.getMendiane() != 0) {
+        Vector3 posMendiane = POS_MENDIANE;
+        DrawModelEx(_mendianeModel, (Vector3){posTile.x + posMendiane.x, posTile.y + posMendiane.y, posTile.z + posMendiane.z}, ROTATION_AXIS_MENDIANE, ROTATION_ANGLE_MENDIANE, SCALE_MENDIANE, WHITE);
+    }
+}
+
+void Gui::Render::displayPhiras(Tile tile)
```
Comment (Njord201) [inline gui/src/Render/Render.cpp:144]:
```suggestion
void Gui::Render::displayPhiras(Tile tile) const
```

File: gui/src/Render/Render.cpp [hunk 7]
```diff
@@ -84,10 +90,93 @@ void Gui::Render::displayDebug(void)
...
+{
+    Vector3 posTile = tile.getPositionIn3DSpace();
+
+    if (tile._inventory.getPhiras() != 0) {
+        Vector3 posPhiras = POS_PHIRAS;
+        DrawModelEx(_phirasModel, (Vector3){posTile.x + posPhiras.x, posTile.y + posPhiras.y, posTile.z + posPhiras.z}, ROTATION_AXIS_PHIRAS, ROTATION_ANGLE_PHIRAS, SCALE_PHIRAS, WHITE);
+    }
+}
+
+void Gui::Render::displaySibur(Tile tile)
```
Comment (Njord201) [inline gui/src/Render/Render.cpp:154]:
```suggestion
void Gui::Render::displaySibur(Tile tile) const
```

File: gui/src/Render/Render.cpp [hunk 8]
```diff
@@ -84,10 +90,93 @@ void Gui::Render::displayDebug(void)
...
+{
+    Vector3 posTile = tile.getPositionIn3DSpace();
+
+    if (tile._inventory.getSibur() != 0) {
+        Vector3 posSibur = POS_SIBUR;
+        DrawModelEx(_siburModel, (Vector3){posTile.x + posSibur.x, posTile.y + posSibur.y, posTile.z + posSibur.z}, ROTATION_AXIS_SIBUR, ROTATION_ANGLE_SIBUR, SCALE_SIBUR, WHITE);
+    }
+}
+
+void Gui::Render::displayThystame(Tile tile)
```
Comment (Njord201) [inline gui/src/Render/Render.cpp:164]:
```suggestion
void Gui::Render::displayThystame(Tile tile) const
```

File: gui/src/Render/Render.cpp [hunk 9]
```diff
@@ -84,10 +90,93 @@ void Gui::Render::displayDebug(void)
...
+{
+    Vector3 posTile = tile.getPositionIn3DSpace();
+
+    if (tile._inventory.getThystame() != 0) {
+        Vector3 posThystame = POS_THYSTAME;
+        DrawModelEx(_thystameModel, (Vector3){posTile.x + posThystame.x, posTile.y + posThystame.y, posTile.z + posThystame.z}, ROTATION_AXIS_THYSTAME, ROTATION_ANGLE_THYSTAME, SCALE_THYSTAME, WHITE);
+    }
+}
+
+void Gui::Render::displayDeraumere(Tile tile)
```
Comment (Njord201) [inline gui/src/Render/Render.cpp:174]:
```suggestion
void Gui::Render::displayDeraumere(Tile tile) const
```

**Review by Njord201 — CHANGES_REQUESTED** [review]

**Review by Njord201 — DISMISSED** [review]

File: gui/include/Assets.hpp [hunk 10]
```diff
@@ -7,5 +7,45 @@
 
 #pragma once
 
-#define MODEL_TILE  "assets/building_platform.glb"
-#define SIZE_TILE   6
+#define MODEL_TILE                  "assets/tile.glb"
+#define MODEL_FOOD                  "assets/food.glb"
```
Comment (theophile-jr) [inline gui/include/Assets.hpp:11]: "PATH_"...

**Review by theophile-jr — DISMISSED** [review]

**Review by Thomaltarix — APPROVED** [review]
Clean one

**Review by theophile-jr — APPROVED** [review]

**Review by Njord201 — APPROVED** [review]

**Comment by theophile-jr** [comment]
Change assets :))

**Pull request merged**